app/api/get-token/route.ts

```diff
@@ -0,0 +1,26 @@
+import { NextResponse } from 'next/server';
+import { getAuthTokenServer } from '@/utils/getAuthTokenServer';
+
+export const GET = async () => {
+    try {
+        const token = await getAuthTokenServer();
+
+        return NextResponse.json({ token }, { status: 200 })
+
+    } catch (err) {
+        console.error('Error fetching token:', err);
+        return NextResponse.json({ error: 'Internal Server Error' }, { status: 405 });
```

Comment:
From the use case here it seems like `500` should be the correct code to use. 

What would be the cases where a token cannot be found? We probably need to handle each of those cases with different error responses.

If the token cannot be found cause a fault on our side, then `500` would be the correct code. But if it cannot be found cause the user isn't signed in, then `401` is probably the correct code.

But most likely the status `405` here is not the right one, since the method is technically allowed, so long as you have a token.

Reply:
Thanks @andrewebdev. I will change it to statusCode of 500.